src/fhirtypes/StructureDefinition.ts

```diff
@@ -85,12 +86,14 @@ export class StructureDefinition {
 
   /**
    * Get the Structure Definition for Structure Definition
-   * @param {ResolveFn} resolve - A function that can resolve a type to a StructureDefinition instance
+   * @param {Fishable} fisher - A fishable implementation for finding definitions and metadata
    * @returns {StructureDefinition} the StructureDefinition of StructureDefinition
    */
-  getOwnStructureDefinition(resolve: ResolveFn = () => undefined): StructureDefinition {
+  getOwnStructureDefinition(fisher: Fishable): StructureDefinition {
     if (this._sdStructureDefinition == null) {
-      this._sdStructureDefinition = resolve('StructureDefinition');
+      this._sdStructureDefinition = StructureDefinition.fromJSON(
+        fisher.fishForFHIR('StructureDefinition', Type.Resource)
+      );
```

Comment:
Similar to the case in `ElementDefinition`, if somehow this isn't found, a TypeError gets thrown.

Reply:
The chances of this happening are extremely small -- and it would be fatal in any case.  So... I'm not terribly concerned.  That said, I have considered if maybe we should make `fromJSON` return `undefined`/`null` when `undefined`/`null` is passed in.  What do you think?  It would still fail somewhere -- just not here.

Reply:
If `fromJSON` returns `undefined`/`null`, the TypeError just gets moved up the stack to `setPropertyOnDefinitionInstance`. Since it's basically impossible for anything useful to be done when `StructureDefinition` is missing, it might be reasonable to have a specific message shown when it isn't found. I think that a case of missing `ElementDefinition` could be handled similarly. If I understand this situation correctly, the most probable causes would be that FHIR wasn't specified as a dependency, or that somehow it got removed from the cache? So a possible message could mention something like checking to make sure that your dependencies in `package.json` are correct.

Reply:
If we don't do so already, we really ought to log an error and fail fast earlier in the process if the user hasn't declared a dependency on FHIR core.  We could also do a spot check for `StructureDefinition` and `ElementDefinition` to ensure that the local FHIR installation isn't horribly wrong.  I'd suggest that as a separate task though.

Reply:
I'll put it on our backlog.